src/cartesi/util.lua

```diff
@@ -231,12 +231,12 @@ function _M.dump_log(log, out)
         -- Otherwise, output access
         elseif ai then
             if ai.proof then indentout(out, indent, "hash %s\n", hexhash8(ai.proof.root_hash)) end
-            local read = accessdatastring(ai.read, ai.log2_size)
+            local read = accessdatastring(ai.read, ai.read_hash, ai.log2_size)
             if ai.type == "read" then
                 indentout(out, indent, "%d: read %s@0x%x(%u): %s\n", i, notes[i] or "", ai.address, ai.address, read)
             else
                 assert(ai.type == "write", "unknown access type")
-                local written = accessdatastring(ai.written, ai.log2_size)
+                local written = accessdatastring(ai.written, ai.written_hash, ai.log2_size)
```

Comment:
`ai.written_hash` is optional. Is this going to crash `accessdatastring`? Or does this log always contain the written hash?

Reply:
`accessdatastring()`  was changed to account for this case.